golang/cosmos/app/app.go

```diff
@@ -546,7 +546,15 @@ func (app *GaiaApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci
 	if err := tmjson.Unmarshal(req.AppStateBytes, &genesisState); err != nil {
 		panic(err)
 	}
-	return app.mm.InitGenesis(ctx, app.appCodec, genesisState)
+	res := app.mm.InitGenesis(ctx, app.appCodec, genesisState)
+
+	// Set Historical infos in InitChain to ignore genesis params
+	// This is needed for IBC connections not to time out easily
+	stakingParams := app.StakingKeeper.GetParams(ctx)
```

Comment:
trying to hunt down the return type of this... got as far as
https://github.com/cosmos/cosmos-sdk/blob/master/x/staking/types/params.go

but from there... ugh... in golang it's not statically evident which names are imported from where?
`Params` is presumably imported from `paramtypes` but one has to look at all the imports to be sure?

Reply:
> but from there... ugh... in golang it's not statically evident which names are imported from where?

Yes, it's statically evident.  If it's not in the format of `<moduleimportstr>.someName` where `moduleimportstr` is named in an import clause, then `someName` must be defined in the current `.go` file or a `.go` file in the same directory as the file that reference it.